packages/babel-plugin-transform-runtime/test/fixtures/runtime-corejs3/instance-inference-optimization/input.js

```diff
@@ -0,0 +1,3 @@
+"".concat(b);
+[].concat(b);
+a.concat(b);
```

Comment:
Where are the type annotations coming from in this test?

The code seems to only work based on type annotations, but I don't see a single annotation here and it still works?

Reply:
`getTypeAnnotation` isn't just an ast property accessor: it runs our rudimental type inference (https://github.com/babel/babel/tree/master/packages/babel-traverse/src/path/inference)

Reply:
Ah interesting. 👍